### PR TITLE
added  check_error_code_on_close

### PR DESCRIPTION
**Pull request by geoffviola**
https://github.com/ros2/launch/issues/41

### CODE REVIEW

File: launch/launch/exit_handler.py [hunk 1]
```diff
@@ -35,6 +35,10 @@ def default_exit_handler(context, ignore_returncode=False):
     if not context.launch_state.teardown:
         context.launch_state.teardown = True
 
+    __default_exit_handler_helper(context)
```
Comment (dirk-thomas) [inline launch/launch/exit_handler.py]: The keyword argument `ignore_returncode` must be passed to the new internal function:

```
__default_exit_handler_helper(context, ignore_returncode=ignore_returncode)
```

File: launch/launch/exit_handler.py [hunk 2]
```diff
@@ -100,3 +104,14 @@ def ignore_signal_exit_handler(context):
             context.task_state.returncode = 0
 
     default_exit_handler(context)
+
+
+def check_error_code_on_close(context):
+    """
+    Succeed if the process shuts down only if the error code is 0.
```
Comment (dirk-thomas) [inline launch/launch/exit_handler.py]: This docblock is unclear to me. It should describe what it is actually doing. E.g. trigger teardown of launch if the task has a non-zero return code

**Comment by mikaelarguedas** [comment]
@geoffviola What is the status on this ? can you rebase this PR and address comments please ? Thanks!

**Comment by geoffviola** [comment]
I probably won't be able to finish this. The use case is less relevant now. The team moved towards bash scripts instead of launch files, which ignores failures. Beta 2 is supposed to have command line parameters.

**Comment by clalancette** [comment]
The original author isn't that interested in this right now, so I'll close this out.  Feel free to reopen if you still want this.

**Pull request closed without merging**